docs/guides/k8s-readiness-liveness.md

```diff
@@ -0,0 +1,182 @@
+
+- The `/ready` endpoint indicates if the service is running and ready to accept traffic
+- The `/status` endpoint indicates if all services in the stack are reachable
+- Custom health checks can be added using the `setCustomHealthCheck` method available on the `globalThis.platformatic` object
+
+## Implementation
+
+### 1. Service Implementation with Custom Health Checks
+
+Create a Platformatic service that implements comprehensive health checks:
+
+```javascript
+import fastify from 'fastify'
+
+export function create () {
+  const app = fastify({ 
+    logger: true, 
+    hostname: process.env.PLT_SERVER_HOSTNAME 
+  })
+
+  // Register custom health check with Platformatic
+  globalThis.platformatic.setCustomHealthCheck(async () => {
+    try {
+      // Add your health checks here
+      // For example:
+      // await Promise.all([
+      //   app.db?.query('SELECT 1'),
+      //   fetch('https://external-service/health')
+      // ])
+      return true
+    } catch (err) {
+      app.log.error(err)
+      return false
+    }
+  })
+
+  return app
+}
+```
+
+### 2. Kubernetes Configuration
+
+Create a Kubernetes deployment configuration that defines the probes:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: demo-readiness-liveness
+  labels:
+    app: demo-readiness-liveness
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: demo-readiness-liveness
+  template:
+    metadata:
+      labels:
+        app: demo-readiness-liveness
+    spec:
+      containers:
+      - name: demo-readiness-liveness
+        image: demo-readiness-liveness:latest
+        ports:
+        - containerPort: 3001
+          name: service
+        - containerPort: 9090
+          name: metrics
+        readinessProbe:
+          httpGet:
+            path: /ready
+            port: 9090
+          initialDelaySeconds: 30
+          periodSeconds: 30
+          failureThreshold: 1
+        livenessProbe:
+          httpGet:
+            path: /status
+            port: 9090
+          initialDelaySeconds: 30
+          periodSeconds: 30
+          failureThreshold: 1
+        resources:
+          requests:
+            memory: "256Mi"
+            cpu: "500m"
+          limits:
+            memory: "512Mi"
+            cpu: "1000m"
+```
+
+Key configuration points:
+
+- **Readiness Probe**: Checks `/ready` endpoint every 30 seconds
+- **Liveness Probe**: Checks `/status` endpoint every 30 seconds
+- Both probes:
+  - `initialDelaySeconds: 30`: Wait 30 seconds before first probe
+  - `periodSeconds: 30`: Check every 30 seconds
+  - `failureThreshold: 1`: Fail after 1 unsuccessful attempt
+
+Please note these values are for demonstration purposes. In a production environment, you should set these values based on your application's characteristics and requirements.
+
+### 3. Environment Configuration
+
+Ensure your service binds to the correct network interface in Kubernetes:
+
+```yaml
+env:
+- name: PLT_SERVER_HOSTNAME
```

Comment:
I think this variable must be used in the `watt.json`, isn't it? Add it to the example as well.